src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php

```diff
@@ -96,4 +98,43 @@ public function seed($class = 'DatabaseSeeder')
 
         return $this;
     }
+
+    /**
+     * Start counting executed database queries.
+     *
+     * @return $this
+     */
+    public function startCountingDatabaseQueries()
+    {
+        if ($this->databaseQueriesExecuted === null) {
+            $this->app->make('db')->listen(function ($query) {
```

Comment:
IMHO this shouldn't be necessary to be done here. The setup for a trait should happen automagically when a trait gets added, see how `\Illuminate\Foundation\Testing\TestCase::setUpTraits` handles this with other traits.

Reply:
Another thing: a project might have multiple database connections, but this trait can only assert the main one.